Library/Homebrew/cmd/update-report.rb

```diff
@@ -749,12 +749,12 @@ def dump(auto_update: false)
       report_all = false
     end
 
-    dump_new_formula_report
-    dump_new_cask_report
+    dump_new_formula_report if ENV["HOMEBREW_NO_UPDATE_REPORT_FORMULAE"].blank?
+    dump_new_cask_report if ENV["HOMEBREW_NO_UPDATE_REPORT_CASKS"].blank?
```

Comment:
```suggestion
    if Homebrew::EnvConfig.no_update_report_new?
      dump_new_formula_report
      dump_new_cask_report
    end
```

Reply:
@eugenesvk check out this suggestion

Reply:
yes, the typed env is much nicer, thanks